components/service/glean/src/main/java/mozilla/components/service/glean/debug/GleanDebugActivity.kt

```diff
@@ -33,16 +40,25 @@ class GleanDebugActivity : Activity() {
 
         // Enable debugging options and start the application.
         intent.extras?.let {
+            var debugId = ""
+
+            // Check for ping debug view tag to apply to the X-Debug-ID header when uploading the
+            // ping to the endpoint
+            if (intent.hasExtra(TAG_DEBUG_VIEW_EXTRA_KEY)) {
+                debugId = intent.getStringExtra(TAG_DEBUG_VIEW_EXTRA_KEY)
```

Comment:
Given the note about unsecured above and debug id constraints specced [in the debug view plan](https://docs.google.com/document/d/1w0aJK6AqqGAo_wr-iXMyBsmLYQTMKc5BXuSVkBS-MxU/edit?ts=5c9385d3#heading=h.1l98j8tqs1il)...
- we should sanity-check the data (filter or reject otherwise)
- we should document the constraints in the README